src/main/java/com/amashchenko/maven/plugin/gitflow/AbstractGitFlowMojo.java

```diff
@@ -316,10 +319,19 @@ private String getCurrentProjectOutputTimestamp() throws MojoFailureException {
      * @throws MojoFailureException
      *             If project loading fails.
      */
-    private MavenProject reloadProject(MavenProject project) throws MojoFailureException {
+    private MavenProject reloadProject(final MavenProject project) throws MojoFailureException {
         try {
-            ProjectBuildingResult result = projectBuilder.build(project.getFile(), mavenSession.getProjectBuildingRequest());
-            return result.getProject();
+            List<ProjectBuildingResult> result = projectBuilder.build(
+                    Collections.singletonList(project.getFile()),
+                    true,
+                    mavenSession.getProjectBuildingRequest());
+
+            for (ProjectBuildingResult projectBuildingResult : result) {
+                MavenProject resultProject = projectBuildingResult.getProject();
+                if (!resultProject.isExecutionRoot()) continue;
+                return resultProject;
+            }
+            throw new NoSuchElementException("None of the  ProjectBuildingResults is for the executionRoot");
```

Comment:
I wonder if this exception message is too "technical" or, in other, too much coupled to the Maven Core API. Maybe:

```suggestion
            throw new NoSuchElementException("No reloaded project appears to be the execution root (" + project.getGroupId() + ":" + project.getArtifactId() + ")");
```

Reply:
I wonder if "Panic this never should have happened" would be a Exception Message just as good.
This mojo must be called in an existing maven Project. Therefore the executionRoot should be part of the List. Or am I wrong?
Nonetheless I'll change the Exception Message